packages/amplify-cli/src/lib/init-steps/preInitSetup.js

```diff
@@ -0,0 +1,121 @@
+const { execSync } = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+async function run(context) {
+  if (context.parameters.options.app) {
+    // Setting up a sample app
+    context.print.warning('Note: Amplify does not have knowledge or necessarily approve of url provided');
+    const repoUrl = context.parameters.options.app;
+    await validateGithubRepo(repoUrl);
+    await cloneRepo(repoUrl);
+    const packageManager = await getPackageManager();
+    await installPackage(packageManager);
+    await setLocalEnvDefaults(context);
+  }
+  return context;
+}
+
+/**
+ * Checks whether a url is a valid address
+ *
+ * @param str the url to validated
+ * @return true if str is a valid address, false otherwise
+ */
+async function isURL(str) {
+  const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
+    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|' + // domain name
+    '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+    '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+    '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
+  return pattern.test(str);
+}
+
+/**
+ * Checks whether a url is a valid remote github repository
+ *
+ * @param repoUrl the url to validated
+ * @throws error if url is not a valid remote github url
+ */
+async function validateGithubRepo(repoUrl) {
+  if (await isURL(repoUrl)) {
+    try {
+      execSync(`git ls-remote ${repoUrl}`, { stdio: 'ignore' });
+    } catch (e) {
+      throw Error('Invalid remote github url');
+    }
+  } else {
+    throw Error('Invalid url');
+  }
+}
+
+/**
+ * Clones repo from url to current directory (must be empty)
+ *
+ * @param repoUrl the url to be cloned
+ */
+async function cloneRepo(repoUrl) {
+  execSync(`git clone ${repoUrl} .`, { stdio: 'inherit' });
+}
+
+/**
+ * Determine the package manager of the current project
+ *
+ * @return 'yarn' if yarn.lock exists, 'npm' otherwise
+*/
+async function getPackageManager() {
+  const yarnLock = './yarn.lock';
+  const yarnLockDir = path.join(process.cwd(), yarnLock);
+  if (fs.existsSync(yarnLockDir)) {
+    return 'yarn';
+  }
+  return 'npm';
+}
+
+/**
+ * Install package using the correct package manager
+ *
+ * @param packageManager either npm or yarn
+ */
+async function installPackage(packageManager) {
+  if (packageManager === 'yarn') {
+    await execSync('yarn install', { stdio: 'inherit' });
+  } else {
+    await execSync('npm install', { stdio: 'inherit' });
+  }
+}
```

Comment:
on windows os, the commands become `npm.cmd` and `yarn.cmd`. you need to do a normalization based on the os. check [here](https://github.com/aws-amplify/amplify-cli/blob/f33c21f44b344b14bc9617eb97c3aa0f8330b138/packages/amplify-frontend-javascript/lib/builder.js#L36)

Reply:
Made this change